src/showingpreviously/cinemas/cineworld.py

```diff
@@ -0,0 +1,105 @@
+import json
+
+from datetime import datetime, timedelta
+import showingpreviously.requests as requests
+from showingpreviously.model import ChainArchiver, CinemaArchiverException, Chain, Cinema, Screen, Film, Showing
+
+
+CINEMAS_API_URL = 'https://www.cineworld.co.uk/uk/data-api-service/v1/quickbook/10108/cinemas/with-event/until/{end_date}'
+SHOWINGS_API_URL = 'https://www.cineworld.co.uk/uk/data-api-service/v1/quickbook/10108/film-events/in-cinema/{cinema_id}/at-date/{date}'
+
+DAYS_AHEAD = 2
+CHAIN = Chain('Cineworld UK')
+
+
+def get_response(url: str) -> requests.Response:
+    r = requests.get(url)
+    if r.status_code != 200:
+        raise CinemaArchiverException(f'Got status code {r.status_code} when fetching URL {url}')
+    return r
+
+
+def get_cinemas_as_dict() -> dict[str, Cinema]:
+    end_date = datetime.now() + timedelta(days=DAYS_AHEAD)
+    url = CINEMAS_API_URL.format(end_date=end_date.strftime('%Y-%m-%d'))
+    r = get_response(url)
+    try:
+        cinemas_data = r.json()
+    except json.JSONDecodeError:
+        raise CinemaArchiverException(f'Error decoding JSON data from URL {url}')
+    cinemas = {}
+    for cinema in cinemas_data['body']['cinemas']:
+        id = cinema['id']
+        name = cinema['displayName']
+        timezone = 'Europe/London'
+        cinemas[id] = Cinema(name, timezone)
+    return cinemas
+
+
+def get_json_attributes(attributes: [str]) -> dict[str, any]:
+    json_attributes = {}
+    for attribute in attributes:
+        if attribute == 'audio-described':
+            json_attributes['audio-described'] = True
+        elif attribute in ['120-fps', '35-mm', '70-mm', '4dx', '4k', 'imax', 'imax-3d', 'imax-3d-vip', 'imax-vr', 'screenx', 'thx', '3d']:
+            if 'format' not in json_attributes:
+                json_attributes['format'] = []
+            json_attributes['format'].append(attribute)
+        elif attribute in ['bengali', 'chinese-st', 'filipino', 'gujarati', 'hindi', 'japanese', 'kannada', 'korean', 'malayalam', 'mandarin', 'marathi', 'nepali', 'punjabi', 'russian', 'spanish', 'tamil', 'telugu', 'urdu', 'vietnamese']:
+            json_attributes['language'] = attribute
+        elif attribute in ['sub-titled', 'subbed', 'korean-sub', 'spanish-sub']:
+            if attribute == 'korean-sub':
+                json_attributes['subtitled'] = 'korean'
+            elif attribute == 'spanish-sub':
+                json_attributes['subtitled'] = 'spanish'
+            else:
+                json_attributes['subtitled'] = True
```

Comment:
Again, a bit concerned about the hard-coding of languages here. Chinese and Japanese, for example, are possible to be subtitled?

Reply:
Their API doesn't seem to think so. I don't have a film to test with, but that is all the attributes their API seems capable of returning. Maybe they add them as they are needed (which would explain the inconsistent naming of them)?